src/display/editor/annotation_editor_layer.js

```diff
@@ -679,14 +679,32 @@ class AnnotationEditorLayer {
 
   /**
    * SelectionChange callback.
-   * @param {Event} _event
+   * @param {Event} event
    */
-  selectionStart(_event) {
-    this.#textLayer?.div.addEventListener(
-      "pointerup",
-      this.#boundPointerUpAfterSelection,
-      { once: true }
-    );
+  selectionStart(event) {
+    if (
+      !this.#textLayer ||
+      event.target.parentElement.closest(".textLayer") !== this.#textLayer.div
+    ) {
+      return;
+    }
+
+    if (this.#uiManager.isShiftKeyDown) {
+      const keyup = () => {
+        if (this.#uiManager.isShiftKeyDown) {
+          return;
+        }
+        this.#textLayer.div.removeEventListener("keyup", keyup);
+        this.pointerUpAfterSelection({});
+      };
+      window.addEventListener("keyup", keyup);
```

Comment:
This event-listener is never being removed, since you register it on the `window` and try to remove it from `this.#textLayer.div`.